### ISSUE TITLE
Intermittant failure to investigate

### ISSUE THREAD

**Issue by hardbyte** [data61/anonlink-entity-service#249]
Another issue that seems to show up occasionally during CI testing e.g. https://magic-ci-atp.it.csiro.au/job/entity-service/job/update-notebooks/7/console

My suspicion/s are the fact there were 0 similarity scores above the threshold, there were 2 worker tasks started when I would have only expected one:

```
[2018-08-13 11:16:41,669: INFO/ForkPoolWorker-321] Queueing run for computation   [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=8bb480b9-5cec-44ff-b85a-4261741055fa task_name=check_for_executable_runs
[2018-08-13 11:16:42,836: INFO/ForkPoolWorker-322] CLK similarity computation scheduled [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=23ff66a8-0ff2-492b-914b-12022d0ac7bf task_name=compute_run
[2018-08-13 11:16:42,837: INFO/ForkPoolWorker-323] CLK similarity computation scheduled [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=cd59f4f1-7b72-4d8d-892b-42cb9af79400 task_name=compute_run
[2018-08-13 11:16:42,841: INFO/ForkPoolWorker-320] Starting comparison of CLKs from data provider ids: 907, 908 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=c78775b3-70a0-4a08-b407-503c28291df3 task_name=compute_similarity
[2018-08-13 11:16:42,841: INFO/ForkPoolWorker-316] Starting comparison of CLKs from data provider ids: 907, 908 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=39976129-53e5-454f-85c8-3b1edc5d888c task_name=compute_similarity
```

The (filtered) logs are:

## Flask App (backend) Logs

```
[2018-08-13 11:16:40 +0000] [13] [INFO] Adding new project to database [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [13] [DEBUG] Trying to connect to postgres db [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [13] [DEBUG] Starting database transaction. Creating project. [entityservice.models.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [13] [DEBUG] New project created in DB      [entityservice.models.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [13] [DEBUG] Creating new data provider entries [entityservice.models.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [13] [DEBUG] Added a dataprovider to db     [entityservice.models.project] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [13] [DEBUG] Added a dataprovider to db     [entityservice.models.project] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [13] [DEBUG] Added data providers           [entityservice.models.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [13] [DEBUG] Committing transaction         [entityservice.models.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5dfa9a6f
[2018-08-13 11:16:40 +0000] [14] [DEBUG] Trying to connect to postgres db [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [DEBUG] Query: select count(*) from projects WHERE project_id = %s [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [DEBUG] checking authorization token to update data [entityservice.views.auth_checks] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [INFO] Receiving CLK data.            [entityservice.views.project] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [INFO] Storing user 907 supplied clks from json [entityservice.views.project] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [INFO] Processed 1000 CLKS            [entityservice.views.project] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [INFO] Uploading 168.95 KiB to object store [entityservice.views.project] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [DEBUG] Trying to connect to postgres db [entityservice.database.util] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [INFO] Adding CLK data to database    [entityservice.database.insertions] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:40 +0000] [14] [DEBUG] Setting dataprovider 907 upload state to True [entityservice.database.insertions] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:41 +0000] [14] [INFO] Job scheduled to handle user uploaded hashes [entityservice.views.project] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=1c06f034
[2018-08-13 11:16:41 +0000] [13] [DEBUG] Trying to connect to postgres db [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [DEBUG] Query: select count(*) from projects WHERE project_id = %s [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [DEBUG] checking authorization token to update data [entityservice.views.auth_checks] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [INFO] Receiving CLK data.            [entityservice.views.project] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [INFO] Storing user 908 supplied clks from json [entityservice.views.project] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [INFO] Processed 1 CLKS               [entityservice.views.project] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [INFO] Uploading 173.00 Byte to object store [entityservice.views.project] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [DEBUG] Trying to connect to postgres db [entityservice.database.util] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [INFO] Adding CLK data to database    [entityservice.database.insertions] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [DEBUG] Setting dataprovider 908 upload state to True [entityservice.database.insertions] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [13] [INFO] Job scheduled to handle user uploaded hashes [entityservice.views.project] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=14d96a67
[2018-08-13 11:16:41 +0000] [14] [DEBUG] Processing request to add a new run [entityservice.views.run.list] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81 run={'threshold': 0.9}
[2018-08-13 11:16:41 +0000] [14] [DEBUG] Trying to connect to postgres db [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [14] [DEBUG] Query: select count(*) from projects WHERE project_id = %s [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [14] [DEBUG] checking authorization of 'result_token' [entityservice.views.auth_checks] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [14] [INFO] Threshold for run is: 0.9      [entityservice.models.run] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [14] [INFO] Created run id                 [entityservice.models.run] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81 rid=ef34e0160feb4c3d400fb759880fa3776efe6007be104694
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [14] [DEBUG] Saving run                     [entityservice.views.run.list] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [14] [DEBUG] Saving run in database         [entityservice.models.run] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81 rid=ef34e0160feb4c3d400fb759880fa3776efe6007be104694
[2018-08-13 11:16:41 +0000] [14] [DEBUG] New run created in DB          [entityservice.models.run] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81 rid=ef34e0160feb4c3d400fb759880fa3776efe6007be104694
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [14] [DEBUG] Expecting 2 parties to upload data. Have received 2 [entityservice.views.run.list] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [14] [INFO] Scheduling task to carry out all runs for project 4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 now [entityservice.views.run.list] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=5f9ede81
[2018-08-13 11:16:41 +0000] [12] [INFO] Request to delete project      [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:41 +0000] [12] [DEBUG] Trying to connect to postgres db [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:41 +0000] [12] [DEBUG] Query: select count(*) from projects WHERE project_id = %s [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:41 +0000] [12] [DEBUG] checking authorization of 'result_token' [entityservice.views.auth_checks] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:41 +0000] [12] [INFO] Authorized request to delete a project resource and all data [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:41 +0000] [12] [INFO] Deleting project resourced from database [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
             [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:41 +0000] [12] [INFO] blooming data file found: {'file': 'raw-clks/4eee42f7849152f7027788d9d2510461629a0e359db06c9e.bin   '} [entityservice.database.selections] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
             [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:41 +0000] [12] [INFO] blooming data file found: {'file': 'raw-clks/bacf9c1e8866bde597d8751baf29a5a4ec229ad62a7ceb2a.bin   '} [entityservice.database.selections] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:41 +0000] [12] [WARNING] Beginning db transaction to remove everything associated with project [entityservice.database.deletion] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=bdea62b9
[2018-08-13 11:16:42 +0000] [12] [ERROR] Error handling request /projects/4a601a951b4a031f0c947f7fd7436bb751ae42b257639727
[2018-08-13 11:16:43 +0000] [12] [INFO] Request to delete project      [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [DEBUG] Trying to connect to postgres db [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [DEBUG] Query: select count(*) from projects WHERE project_id = %s [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [DEBUG] checking authorization of 'result_token' [entityservice.views.auth_checks] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [INFO] Authorized request to delete a project resource and all data [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [INFO] Deleting project resourced from database [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
         [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
             [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [INFO] blooming data file found: {'file': 'raw-clks/4eee42f7849152f7027788d9d2510461629a0e359db06c9e.bin   '} [entityservice.database.selections] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
             [entityservice.database.util] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [INFO] blooming data file found: {'file': 'raw-clks/bacf9c1e8866bde597d8751baf29a5a4ec229ad62a7ceb2a.bin   '} [entityservice.database.selections] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [WARNING] Beginning db transaction to remove everything associated with project [entityservice.database.deletion] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [INFO] Committing removal of project resource [entityservice.database.deletion] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [INFO] Object store files associated with project: [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [INFO] Deleting raw-clks/4eee42f7849152f7027788d9d2510461629a0e359db06c9e.bin    [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
[2018-08-13 11:16:43 +0000] [12] [INFO] Deleting raw-clks/bacf9c1e8866bde597d8751baf29a5a4ec229ad62a7ceb2a.bin    [entityservice.views.project] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 request=4c2df4df
```

## Worker Logs

```
[2018-08-13 11:16:41,061: INFO/ForkPoolWorker-322] Handling user uploaded file    [celery] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=1944b63d-8efa-4cf8-82fe-e3c586d77860 task_name=handle_raw_upload
[2018-08-13 11:16:41,070: INFO/ForkPoolWorker-322] Expecting to handle 1000 hashes [celery] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=1944b63d-8efa-4cf8-82fe-e3c586d77860 task_name=handle_raw_upload
[2018-08-13 11:16:41,081: INFO/ForkPoolWorker-322] Caching pickled clk data       [celery] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=1944b63d-8efa-4cf8-82fe-e3c586d77860 task_name=handle_raw_upload
[2018-08-13 11:16:41,094: INFO/ForkPoolWorker-322] Processed 1000 hashes          [celery] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=1944b63d-8efa-4cf8-82fe-e3c586d77860 task_name=handle_raw_upload
[2018-08-13 11:16:41,101: INFO/ForkPoolWorker-322] Uploading binary packed clks to object store. Size: 130.86 KiB [celery] dp_id=907 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=1944b63d-8efa-4cf8-82fe-e3c586d77860 task_name=handle_raw_upload
[2018-08-13 11:16:41,290: INFO/ForkPoolWorker-323] Handling user uploaded file    [celery] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=ce85d873-3120-433c-9d85-2a62e3c84688 task_name=handle_raw_upload
[2018-08-13 11:16:41,298: INFO/ForkPoolWorker-323] Expecting to handle 1 hashes   [celery] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=ce85d873-3120-433c-9d85-2a62e3c84688 task_name=handle_raw_upload
[2018-08-13 11:16:41,309: INFO/ForkPoolWorker-323] Caching pickled clk data       [celery] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=ce85d873-3120-433c-9d85-2a62e3c84688 task_name=handle_raw_upload
[2018-08-13 11:16:41,310: INFO/ForkPoolWorker-323] Processed 1 hashes             [celery] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=ce85d873-3120-433c-9d85-2a62e3c84688 task_name=handle_raw_upload
[2018-08-13 11:16:41,312: INFO/ForkPoolWorker-323] Uploading binary packed clks to object store. Size: 134.00 Byte [celery] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=ce85d873-3120-433c-9d85-2a62e3c84688 task_name=handle_raw_upload
[2018-08-13 11:16:41,406: INFO/ForkPoolWorker-323] All parties' data present. Scheduling any queued runs [celery] dp_id=908 pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=ce85d873-3120-433c-9d85-2a62e3c84688 task_name=handle_raw_upload
[2018-08-13 11:16:41,454: INFO/ForkPoolWorker-320] Checking for runs that need to be executed [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=1c0abf7b-652d-4905-82b3-5cea0ab9d2a4 task_name=check_for_executable_runs
[2018-08-13 11:16:41,502: INFO/ForkPoolWorker-320] Creating tasks for 1 created runs for project 4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=1c0abf7b-652d-4905-82b3-5cea0ab9d2a4 task_name=check_for_executable_runs
[2018-08-13 11:16:41,502: INFO/ForkPoolWorker-320] Queueing run for computation   [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=1c0abf7b-652d-4905-82b3-5cea0ab9d2a4 task_name=check_for_executable_runs
[2018-08-13 11:16:41,506: INFO/ForkPoolWorker-321] Checking for runs that need to be executed [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=8bb480b9-5cec-44ff-b85a-4261741055fa task_name=check_for_executable_runs
[2018-08-13 11:16:41,669: INFO/ForkPoolWorker-321] Creating tasks for 1 created runs for project 4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 task_id=8bb480b9-5cec-44ff-b85a-4261741055fa task_name=check_for_executable_runs
[2018-08-13 11:16:41,669: INFO/ForkPoolWorker-321] Queueing run for computation   [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=8bb480b9-5cec-44ff-b85a-4261741055fa task_name=check_for_executable_runs
[2018-08-13 11:16:42,836: INFO/ForkPoolWorker-322] CLK similarity computation scheduled [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=23ff66a8-0ff2-492b-914b-12022d0ac7bf task_name=compute_run
[2018-08-13 11:16:42,837: INFO/ForkPoolWorker-323] CLK similarity computation scheduled [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=cd59f4f1-7b72-4d8d-892b-42cb9af79400 task_name=compute_run
[2018-08-13 11:16:42,841: INFO/ForkPoolWorker-320] Starting comparison of CLKs from data provider ids: 907, 908 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=c78775b3-70a0-4a08-b407-503c28291df3 task_name=compute_similarity
[2018-08-13 11:16:42,841: INFO/ForkPoolWorker-316] Starting comparison of CLKs from data provider ids: 907, 908 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=39976129-53e5-454f-85c8-3b1edc5d888c task_name=compute_similarity
[2018-08-13 11:16:42,847: INFO/ForkPoolWorker-320] Computing similarity for 1000 x 1 entities [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=c78775b3-70a0-4a08-b407-503c28291df3 task_name=compute_similarity
[2018-08-13 11:16:42,847: INFO/ForkPoolWorker-316] Computing similarity for 1000 x 1 entities [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=39976129-53e5-454f-85c8-3b1edc5d888c task_name=compute_similarity
[2018-08-13 11:16:42,848: INFO/ForkPoolWorker-320] Computing similarity using greedy method [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=c78775b3-70a0-4a08-b407-503c28291df3 task_name=compute_similarity
[2018-08-13 11:16:42,848: INFO/ForkPoolWorker-316] Computing similarity using greedy method [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=39976129-53e5-454f-85c8-3b1edc5d888c task_name=compute_similarity
[2018-08-13 11:16:42,849: INFO/ForkPoolWorker-316] Chunks will contain 1000 entities per task [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=39976129-53e5-454f-85c8-3b1edc5d888c task_name=compute_similarity
[2018-08-13 11:16:42,849: INFO/ForkPoolWorker-320] Chunks will contain 1000 entities per task [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=c78775b3-70a0-4a08-b407-503c28291df3 task_name=compute_similarity
[2018-08-13 11:16:42,850: INFO/ForkPoolWorker-320] Chunking into 1 computation tasks each with (at most) 1000 entities. [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=c78775b3-70a0-4a08-b407-503c28291df3 task_name=compute_similarity
[2018-08-13 11:16:42,850: INFO/ForkPoolWorker-316] Chunking into 1 computation tasks each with (at most) 1000 entities. [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=39976129-53e5-454f-85c8-3b1edc5d888c task_name=compute_similarity
[2018-08-13 11:16:42,915: INFO/ForkPoolWorker-317] run=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 Comparisons: 1000, Links above threshold: 0 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=9d7d41cf-8b59-45c9-aec5-6ea5e20f92cb task_name=compute_filter_similarity
[2018-08-13 11:16:42,915: INFO/ForkPoolWorker-317] Prep: 0.035 + 0.008, Solve: 0.007, Progress: 0.001, Offset: 0.000, Save: 0.000, Total: 0.051 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=9d7d41cf-8b59-45c9-aec5-6ea5e20f92cb task_name=compute_filter_similarity
[2018-08-13 11:16:42,915: INFO/ForkPoolWorker-322] run=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 Comparisons: 1000, Links above threshold: 0 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=223364e5-8f12-4257-bc2e-8d2fb445b9ee task_name=compute_filter_similarity
[2018-08-13 11:16:42,915: INFO/ForkPoolWorker-322] Prep: 0.033 + 0.009, Solve: 0.007, Progress: 0.001, Offset: 0.000, Save: 0.000, Total: 0.050 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=223364e5-8f12-4257-bc2e-8d2fb445b9ee task_name=compute_filter_similarity
[2018-08-13 11:16:42,952: INFO/ForkPoolWorker-322] Similarity score results are 0.00 Bit [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=33cbbc91-441d-41c5-9f4a-fbe404581b85 task_name=aggregate_filter_chunks
[2018-08-13 11:16:42,953: INFO/ForkPoolWorker-311] Similarity score results are 0.00 Bit [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=f3de595f-44e5-47b0-9fa8-ecb2ce8774d0 task_name=aggregate_filter_chunks
[2018-08-13 11:16:42,957: INFO/ForkPoolWorker-322] Storing similarity score results in CSV file: similarity-scores/ef34e0160feb4c3d400fb759880fa3776efe6007be104694.csv [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=33cbbc91-441d-41c5-9f4a-fbe404581b85 task_name=aggregate_filter_chunks
[2018-08-13 11:16:42,959: INFO/ForkPoolWorker-311] Storing similarity score results in CSV file: similarity-scores/ef34e0160feb4c3d400fb759880fa3776efe6007be104694.csv [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=f3de595f-44e5-47b0-9fa8-ecb2ce8774d0 task_name=aggregate_filter_chunks
[2018-08-13 11:16:43,095: INFO/ForkPoolWorker-322] Calculating the optimal mapping from similarity matrix [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=f29d0a35-dd85-4199-9cc9-3cee89c73405 task_name=solver_task
[2018-08-13 11:16:43,096: INFO/ForkPoolWorker-322] Entity mapping has been computed [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=f29d0a35-dd85-4199-9cc9-3cee89c73405 task_name=solver_task
[2018-08-13 11:16:43,139: INFO/ForkPoolWorker-316] Calculating the optimal mapping from similarity matrix [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=726b96fa-5699-4de3-8154-0718a3580b4c task_name=solver_task
[2018-08-13 11:16:43,139: INFO/ForkPoolWorker-316] Entity mapping has been computed [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=726b96fa-5699-4de3-8154-0718a3580b4c task_name=solver_task
[2018-08-13 11:16:43,160: INFO/ForkPoolWorker-325] Mapping result saved to db with result id 110 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=0c50b76c-a53b-4142-980e-4a8edbfa2df9 task_name=save_and_permute
[2018-08-13 11:16:43,202: INFO/ForkPoolWorker-317] Mapping result saved to db with result id 111 [celery] pid=4a601a951b4a031f0c947f7fd7436bb751ae42b257639727 run_id=ef34e0160feb4c3d400fb759880fa3776efe6007be104694 task_id=5c4d3411-2003-43ea-bf83-ce3c16ee9c8a task_name=save_and_permute
```

## Database

```
626-2018-08-13 11:16:33.196 UTC [11143] LOG:  unexpected EOF on client connection with an open transaction
627-2018-08-13 11:16:40.278 UTC [11283] LOG:  unexpected EOF on client connection with an open transaction
628-2018-08-13 11:16:42.669 UTC [11389] ERROR:  deadlock detected
629-2018-08-13 11:16:42.669 UTC [11389] DETAIL:  Process 11389 waits for ShareLock on transaction 4651; blocked by process 11387.
630-    Process 11387 waits for ShareLock on transaction 4652; blocked by process 11389.
631-    Process 11389:
632-                DELETE FROM projects
633:                WHERE project_id = '4a601a951b4a031f0c947f7fd7436bb751ae42b257639727'
634-
635-    Process 11387:
636-                    UPDATE runs SET
637-                      stage = stage + 1
638-                    WHERE
639-                      run_id = 'ef34e0160feb4c3d400fb759880fa3776efe6007be104694'
640-
641-2018-08-13 11:16:42.669 UTC [11389] HINT:  See server log for query details.
642-2018-08-13 11:16:42.669 UTC [11389] CONTEXT:  while deleting tuple (0,94) in relation "runs"
643-    SQL statement "DELETE FROM ONLY "public"."runs" WHERE $1 OPERATOR(pg_catalog.=) "project""
644-2018-08-13 11:16:42.669 UTC [11389] STATEMENT:
645-                DELETE FROM projects
646:                WHERE project_id = '4a601a951b4a031f0c947f7fd7436bb751ae42b257639727'
647-
648-2018-08-13 11:16:42.866 UTC [11339] LOG:  unexpected EOF on client connection with an open transaction
649-2018-08-13 11:16:42.950 UTC [11310] LOG:  unexpected EOF on client connection with an open transaction
650-2018-08-13 11:16:43.450 UTC [11365] LOG:  unexpected EOF on client connection with an open transaction
```

Aha! Link: https://csiro.aha.io/features/ANONLINK-12
